renovate.json

```
@@ -0,0 +1,73 @@
+{
+  "extends": [
+    "config:js-lib",
+    ":pinOnlyDevDependencies",
```

`:pinOnlyDevDependencies` will lock `devDependencies` versions. For details RE: why renovate suggests doing this, they have a writeup [here](https://renovatebot.com/docs/dependency-pinning/)

If we want to leave SemVer ranges alone and just update the lock file, you can accept this suggestion 👇
```suggestion
    ":preserveSemverRanges",
```

---

I'm generally more comfortable with updating the lockfile, since we can't pin transitive deps anyways, and we have tests which run with dependencies locked and then with them unlocked which helps us narrow down failures due to dependencies shifting. 

This _would_ help us narrow down dependency vs devDependency bugs a bit, but would also add noise in other ways I think (e.g. if a devDependency's transitive dep updated, could still cause errors)